### PR TITLE
Convert help info context to zustand

### PR DESCRIPTION
**Pull request by foxymiles**
**User-Facing Changes**
None

**Description**
It looks like the `HelpInfo` context was another context we've hacked with a hand rolled listener system to avoid the performance pitfalls of React contexts. We can replace this with Zustand.

### CODE REVIEW

File: packages/studio-base/src/panels/NodePlayground/index.tsx [hunk 1]
```diff
@@ -129,8 +129,10 @@ function buildSettingsStree(config: Config): SettingsTreeNode {
   };
 }
 
+const selectSetHelpInfo = (store: HelpInfoStore) => store.setHelpInfo;
```
Comment (jtbandes) [inline packages/studio-base/src/panels/NodePlayground/index.tsx:132]: Does zustand recommend hoisting these selectors so they don't change, or are they ok with them changing? Are the selectors allowed to depend on other props?

Comment (foxymiles) [reply packages/studio-base/src/panels/NodePlayground/index.tsx:132]: There's a marginal performance benefit to hoisting them this way. 

https://github.com/pmndrs/zustand/discussions/658

The selector can be any function you want but you'd probably want to `useCallback` if you want it to be as stable as possible.

Comment (jtbandes) [reply packages/studio-base/src/panels/NodePlayground/index.tsx:132]: We used to have self-memoizing selectors like that from reselect. But we have pretty much fully transitioned to using React's memoization hooks/primitives.

File: packages/studio-base/src/components/HelpSidebar/index.stories.tsx [hunk 2]
```diff
@@ -44,30 +41,6 @@ class MockPanelCatalog implements PanelCatalog {
   }
 }
 
-const MockHelpInfoContext = createContext<IHelpInfo | undefined>(undefined);
-MockHelpInfoContext.displayName = "MockHelpInfoContext";
-function MockHelpInfoProvider({ children }: React.PropsWithChildren<unknown>): JSX.Element {
-  const helpInfo = useRef<HelpInfo>({ title: "Some title", content: <>Some help content</> });
```
Comment (jtbandes) [inline packages/studio-base/src/components/HelpSidebar/index.stories.tsx:50]: Looks like this is removing the dummy/fixture data. Why didn't that result in a change in storybook?

Comment (foxymiles) [reply packages/studio-base/src/components/HelpSidebar/index.stories.tsx:50]: Looks like that content was never actually rendered in the story.

Comment (jtbandes) [reply packages/studio-base/src/components/HelpSidebar/index.stories.tsx:50]: Is the story providing any value if it doesn't actually show the help info? Maybe we should improve or delete it?

Comment (foxymiles) [reply packages/studio-base/src/components/HelpSidebar/index.stories.tsx:50]: It is showing other content, just not this initial value in a mock provider which we don't need.

**Review by jtbandes — APPROVED** [review]

**Pull request merged**